src/extension.ts

```diff
@@ -83,44 +84,29 @@ function showErrorStatusBar() {
     statusBarItem.show();
 }
 
-function executeTypeProf(folder: vscode.WorkspaceFolder, arg: String): child_process.ChildProcessWithoutNullStreams {
+function executeTypeProf(folder: vscode.WorkspaceFolder, arg: string): child_process.ChildProcessWithoutNullStreams {
     const configuration = vscode.workspace.getConfiguration(CONFIGURATION_ROOT_SECTION);
     const customServerPath = configuration.get<string | null>('server.path');
     const cwd = folder.uri.fsPath;
 
     let cmd: string;
-    if (existsSync(`${cwd}/bin/typeprof`)) {
-        cmd = './bin/typeprof';
+    const cmdArgs: string[] = [];
+
+    const binCmd = path.join(cwd, 'bin', 'typeprof');
+    if (existsSync(binCmd)) {
+        cmd = binCmd;
     } else if (customServerPath) {
         cmd = customServerPath;
-    } else if (existsSync(`${cwd}/Gemfile`)) {
-        cmd = 'bundle exec typeprof';
+    } else if (existsSync(path.join(cwd, 'Gemfile'))) {
+        cmd = 'bundle';
+        cmdArgs.push('exec', 'typeprof');
     } else {
         cmd = 'typeprof';
     }
-    cmd = cmd + ' ' + arg;
-
-    const shell = process.env.SHELL;
-    let typeprof: child_process.ChildProcessWithoutNullStreams;
-    if (shell && (shell.endsWith('bash') || shell.endsWith('zsh') || shell.endsWith('fish'))) {
-        const args: string[] = [];
-        if (shell.endsWith('zsh')) {
-            // As the recommended way, initialization commands for rbenv are written in ".zshrc".
-            // However, it's not loaded on the non-interactive shell.
-            // Thus, we need to run this command as the interactive shell.
-            // FYI: https://zsh.sourceforge.io/Guide/zshguide02.html
-            args.push('-i');
-        }
-        args.push('-l', '-c', cmd);
-        typeprof = child_process.spawn(shell, args, { cwd });
-    } else if (process.platform === 'win32') {
-        typeprof = child_process.spawn(process.env.SYSTEMROOT + '\\System32\\cmd.exe', ['/c', cmd], { cwd });
-    } else {
-        const cmds = cmd.split(' ');
-        typeprof = child_process.spawn(cmds[0], cmds.slice(1), { cwd });
-    }
```

Comment:
[note] I try looking into the `shell` option more (Node.js internal). See the code in Node.js:

```js
  if (options.shell) {
    validateArgumentNullCheck(options.shell, 'options.shell');
    const command = ArrayPrototypeJoin([file, ...args], ' ');
    // Set the shell, switches, and commands.
    if (process.platform === 'win32') {
      if (typeof options.shell === 'string')
        file = options.shell;
      else
        file = process.env.comspec || 'cmd.exe';
      // '/d /s /c' is used only for cmd.exe.
      if (RegExpPrototypeExec(/^(?:.*\\)?cmd(?:\.exe)?$/i, file) !== null) {
        args = ['/d', '/s', '/c', `"${command}"`];
        windowsVerbatimArguments = true;
      } else {
        args = ['-c', command];
      }
    } else {
      if (typeof options.shell === 'string')
        file = options.shell;
      else if (process.platform === 'android')
        file = '/system/bin/sh';
      else
        file = '/bin/sh';
      args = ['-c', command];
    }
  }
```

https://github.com/nodejs/node/blob/49412c9632dc697ff52a3f731c578afc26c4654b/lib/child_process.js#L617-L642

This Node.js code might be helpful if we would not use the `shell` option, especially for Windows or Android.

---

Now I understand the need to work around shells (e.g. `zsh -i -l -c "..."`), so I'll close this PR later (I'll keep it open for a while for other people who encounter similar problems).